MSAL/src/native_auth/public/state_machine/error/RetrieveAccessTokenError.swift

```diff
@@ -31,6 +31,7 @@ public class RetrieveAccessTokenError: MSALNativeAuthError {
         case browserRequired
         case refreshTokenExpired
         case generalError
+        case invalidScope
```

Comment:
We need to add a new public utility function `isInvalidScope()` so the developers can check for this.

Remember to add a new test to the `RetrieveAccessTokenErrorTests` to test this new method too.